src/main/java/org/jenkinsci/plugins/workflow/cps/DSL.java

```diff
@@ -455,22 +501,37 @@ private static int preallocatedHashmapCapacity(int elementsToHold) {
         }
     }
 
+    // TODO: add java doc
     static class NamedArgsAndClosure {
         final Map<String,Object> namedArgs;
         final Closure body;
+        final List<String> msgs;
+        final Set<String> interpolatedStrings;
 
-        private NamedArgsAndClosure(Map<?,?> namedArgs, Closure body) {
+        private NamedArgsAndClosure(Map<?,?> namedArgs, Closure body, Set<String> interpolatedStrings) {
             this.namedArgs = new LinkedHashMap<>(preallocatedHashmapCapacity(namedArgs.size()));
             this.body = body;
+            this.msgs = new ArrayList<>();
+            this.interpolatedStrings = interpolatedStrings;
 
             for (Map.Entry<?,?> entry : namedArgs.entrySet()) {
                 String k = entry.getKey().toString().intern(); // coerces GString and more
-                Object v = flattenGString(entry.getValue());
+                Object v = flattenGString(entry.getValue(), interpolatedStrings);
                 this.namedArgs.put(k, v);
             }
         }
     }
 
+//    static class InterpolatedUninstantiatedDescribable {
```

Comment:
At first I tried wrapping the `UninstantiatedDescribable` in a simple class where I could add the interpolated strings. However, that failed when trying to coerce the class in `DescribableModel`
ex:
`java.lang.ClassCastException: org.jenkinsci.plugins.credentialsbinding.impl.BindingStep.bindings expects class org.jenkinsci.plugins.credentialsbinding.MultiBinding but received class org.jenkinsci.plugins.workflow.cps.DSL$InterpolatedUninstantiatedDescribable`

Reply:
removed